frontend/app/mocks/subscription-api.server.ts

```diff
@@ -258,3 +302,34 @@ export function getSubscriptionApiMockHandlers() {
     }),
   ];
 }
+
+/**
+ * Retrieves a user entity based on the provided user ID.
+ *
+ * @param id - The user ID to look up in the database.
+ * @returns The user entity if found, otherwise throws a 404 error.
+ */
+function getUserEntity(id: string | readonly string[]) {
+  const parsedRaoidUserId = z.string().uuid().safeParse(id);
+  const userEntity = !parsedRaoidUserId.success
+    ? undefined
+    : db.user.findFirst({
+        where: { id: { equals: parsedRaoidUserId.data } },
+      });
```

Comment:
This function is finding by user id and not RAOIDC user ID

```suggestion
  const parsedUserId = z.string().uuid().safeParse(id);
  const userEntity = !parsedUserId.success
    ? undefined
    : db.user.findFirst({
        where: { id: { equals: parsedUserId.data } },
      });
```